File: apple/Sources/ReactNativeWatchOS/Modifiers/AllowsHitTesting.swift
import SwiftUI

/// SwiftUI `.allowsHitTesting(_:)`. When `false`, taps/gestures pass through
/// to whatever is behind the view. Defaults to `true` when the param is
/// missing or malformed.
enum RNWAllowsHitTestingModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("allowsHitTesting") { view, params, _ in
            AnyView(view.allowsHitTesting(params.bool("value") ?? true))
        }
    }
}

File: src/modifiers/allowsHitTesting.ts
import { createModifier } from './createModifier';

/// SwiftUI `.allowsHitTesting(_:)`. When `false`, the view is invisible to
/// taps and gestures, which pass through to whatever is behind it. Pass
/// `true` (the default) to restore normal hit testing.
export function allowsHitTesting(value: boolean) {
  return createModifier('allowsHitTesting', { value });
}
